numba/np/arraymath.py

```diff
@@ -3480,49 +3480,109 @@ def impl(arr, axis=None):
-            arr = np.ravel(np.asarray(arr))
-            N = arr.size
+                keep = np.ones(N, dtype=np.bool_)
+                obj = handler(obj)
+                keep[obj] = False
+                return arr[keep]
+        else:
+            ndim = arr.ndim
+
+            def np_delete_impl(arr, obj, axis=None):
+                if axis > arr.ndim - 1:
+                    raise IndexError("axis must less than ndim of arr")
+
+                shape = arr.shape
+                keep = np.ones(shape, dtype=np.bool_)
+                slobj = build_full_slice_tuple(ndim)
+                if obj_is_slice:
+                    slobj = tuple_setitem(slobj, axis, obj)
+                    keep[slobj] = False
+                    obj_length = (obj.stop - obj.start) / obj.step
+                else:
+                    for pos in obj:
+                        if (pos < 0):
+                            pos += shape[axis]
+                        slobj = tuple_setitem(slobj, axis, slice(pos, pos + 1))
+                        keep[slobj] = False
+                    obj_length = len(obj)
+
+                new_shape = tuple_setitem(shape, axis, shape[axis] - obj_length)
+                arr = np.ravel(arr)
+                keep = np.ravel(keep)
+                res = arr[keep].reshape(new_shape)
+                return res
 
-            keep = np.ones(N, dtype=np.bool_)
-            obj = handler(obj)
-            keep[obj] = False
-            return arr[keep]
         return np_delete_impl
 
     else: # scalar value
         if not isinstance(obj, types.Integer):
             raise TypingError('obj should be of Integer dtype')
 
-        def np_delete_scalar_impl(arr, obj):
-            arr = np.ravel(np.asarray(arr))
-            N = arr.size
-            pos = obj
+        if axis in (None, types.none):
+            def np_delete_scalar_impl(arr, obj, axis=None):
+                arr = np.ravel(np.asarray(arr))
+                N = arr.size
+                pos = obj
+
+                if (pos < -N or pos >= N):
+                    raise IndexError('obj must be less than the len(arr)')
+                    # NumPy raises IndexError: index 'i' is out of
+                    # bounds for axis 'x' with size 'n'
+
+                if (pos < 0):
+                    pos += N
+
+                return np.concatenate((arr[:pos], arr[pos + 1:]))
+        else:
+            def np_delete_scalar_impl(arr, obj, axis=None):
+                if axis > arr.ndim - 1:
+                    raise IndexError("axis must less than ndim of arr")
+
+                N = arr.shape[axis]
+                if (obj < -N or obj >= N):
+                    raise IndexError('obj must be less than arr.shape[axis]')
+                if (obj < 0):
+                    obj += N
+
+                new_shape = tuple_setitem(arr.shape, axis,arr.shape[axis] - 1)
+                new = np.empty(new_shape, arr.dtype)
+
+                slobj = build_full_slice_tuple(arr.ndim)
```

Comment:
I think this cannot have runtime integers as arguments. (So it will probably fail, since `arr.ndim` is an integer that is considered as not fixed during runtime).